lib/pco_api/record_association.ex

```diff
@@ -4,16 +4,19 @@ defmodule PcoApi.RecordAssociation do
       def unquote(:"#{name}")([]), do: []
       def unquote(:"#{name}")(%PcoApi.Record{} = record) do
         # TODO: error handling when the association link doesn't exist
-        request(:get, record.links[Atom.to_string(unquote(name))], [])
-        |> Enum.map(fn(%{"attributes" => attrs, "id" => id, "links" => links, "type" => type}) ->
-          %PcoApi.Record{attributes: attrs, id: id, links: links, type: type}
-        end)
+        unquote(:"#{name}")(request(:get, record.links[Atom.to_string(unquote(name))], []))
       end
       def unquote(:"#{name}")([%PcoApi.Record{} = first | rest]) do
         [unquote(:"#{name}")(first) | unquote(:"#{name}")(rest)]
         |> List.flatten
         |> Enum.dedup
       end
+      def unquote(:"#{name}")(%{"attributes" => attrs, "id" => id, "links" => links, "type" => type}) do
+        %PcoApi.Record{attributes: attrs, id: id, links: links, type: type}
+      end
+      def unquote(:"#{name}")(list) when is_list(list) do
+        list |> Enum.map(&(unquote(:"#{name}")(&1)))
+      end
```

Comment:
It's still a little hard to read the metaprogramming in Elixir, but are you doing anything in this file other than just breaking the function into smaller functions?